letsencrypt/cli.py

```diff
@@ -692,7 +695,26 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):
         " used to create obtain or most recently successfully renew each"
         " certificate lineage. You can try it with `--dry-run` first. For"
         " more fine-grained control, you can renew individual lineages with"
-        " the `certonly` subcommand.")
+        " the `certonly` subcommand. Hooks are available to run commands "
+        " before and after renewal; see XXX for more information on these.")
+
+    helpful.add(
+        "renew", "--pre-hook",
+        help="Command to be run in a shell before obtaining any certificates. Intended"
+        " primarily for renewal, where it can be used to temporarily shut down a"
+        " webserver that might conflict with the standalone plugin. This will "
+        " only be called if a certificate is actually to be obtained/renewed. ")
+    helpful.add(
+        "renew", "--post-hook",
+        help="Command to be run in a shell after attempting to obtain/renew "
+        " certificates. Can be used to deploy renewed certificates, or to restart"
+        " any servers that were stopped by --pre-hook.")
+    helpful.add(
+        "renew", "--renew-hook",
+        help="Command to be run in a shell once for each renewed certificate."
+        "For this command, the shell variable $RENEWED_LINEAGE will point to the"
+        "config live subdirectory containing the new certs and keys; the shell variable "
```

Comment:
As a further concern, I mentioned elsewhere the possibility that after the storage reorganization there will no longer be a single live dir containing all four PEM objects. If that happens, people's existing hook scripts could break, especially if they try to reconstruct the four paths by just concatenating `something.pem` to the `RENEWED_LINEAGE` variable. Although this is super-annoying, I think it might be good future-proofing to instead provide four separate variables (e.g. `RENEWED_LINEAGE_CERT`, `RENEWED_LINEAGE_CHAIN`, `RENEWED_LINEAGE_FULLCHAIN`, and `RENEWED_LINEAGE_PRIVKEY`), because this can still be correct and the variables can still be used in the same way following the reorganization. This might encourage/force users to write safer hook scripts, in the sense of more reorganization-tolerant ones.